sdk/python/lib/pulumi/runtime/rpc.py

```diff
@@ -166,6 +175,10 @@
 
         return obj
 
+    # Ensure that we have a value that Protobuf understands.
+    if not isLegalProtobufValue(value):
+        raise ValueError(f"unexpected input of type {type(value).__name__}")
```

Comment:
Should we be a little more descriptive?  Perhaps “inputs must be primitives, lists or dicts”?

Reply:
Yeah, I wasn't sure how verbose we wanted to be. Do you think it's misleading to omit "Outputs, Awaitables, Resources, Assets, or Archives" from that message? 
Is there a docs page we could point to?

Reply:
Ahh - good point that we would need to include these.  I'm okay leaving as you have it for now.  Ideally we would add a docs page to talk about the "Pulumi Object Model" or something like that to describe what the shape of these kinds of values can be.  And then link to that from here once we have it.